source/grid/grid_out.cc

```diff
@@ -1398,10 +1383,7 @@ GridOut::write_xfig(const Triangulation<2> &tria,
   // on finer levels. Level 0
   // corresponds to a depth of 900,
   // each level subtracting 1
-  Triangulation<dim, spacedim>::cell_iterator       cell = tria.begin();
-  const Triangulation<dim, spacedim>::cell_iterator end  = tria.end();
-
-  for (; cell != end; ++cell)
+  for (const auto &cell : tria.active_cell_iterators())
```

Comment:
Here's your bug. You should loop over all, not just the active cells.

Reply:
Thanks!